src/tests/dotnet-trace/ProviderParsing.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Xunit;
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+    public class ProviderParsingTests
+    {
+        [Theory]
+        [InlineData("VeryCoolProvider:0x1:5:FilterAndPayloadSpecs=\"QuotedValue\"")]
+        [InlineData("VeryCoolProvider:1:5:FilterAndPayloadSpecs=\"QuotedValue\"")]
+        public void ValidProvider_CorrectlyParses(string providerToParse)
+        {
+            List<Provider> parsedProviders = Extensions.ToProviders(providerToParse);
+            Assert.True(parsedProviders.Count == 1);
+            Provider provider = parsedProviders.First();
+            Assert.True(provider.Name == "VeryCoolProvider");
+            Assert.True(provider.Keywords == 1);
+            Assert.True(provider.EventLevel == System.Diagnostics.Tracing.EventLevel.Verbose);
+            Assert.True(provider.FilterData == "FilterAndPayloadSpecs=\"QuotedValue\"");
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData(",")]
+        public void EmptyProvider_CorrectlyThrows(string providerToParse)
+        {
+            Assert.Throws<ArgumentNullException>(() => Extensions.ToProviders(providerToParse));
+        }
+
+        [Theory]
+        // [InlineData("VeryCool;Provider:0x1:5:FilterAndPayloadSpecs=\"QuotedValue\"")]
```

Comment:
Remove instead of commenting out?

Reply:
Ah, yup!  Meant to remove those.  We should add some invalid character checks to provider creation parsing.  I might adjust the behavior as a result of these tests.